packages/mgt-components/src/graph/graph.files.ts

```diff
@@ -201,3 +201,299 @@ export async function getUserInsightsDriveItemById(
+  return result || null;
+}
+
+export async function getFilesByListQuery(graph: IGraph, listQuery: string): Promise<DriveItem[]> {
+  const scopes = ['files.read', 'sites.read.all'];
+  let request;
+  let files;
+
+  try {
+    request = await graph.api(listQuery).middlewareOptions(prepScopes(...scopes));
+    files = await getFilesPageIterator(graph, request);
+  } catch {}
+  return files || null;
+}
+
+export async function getFilesByQueries(graph: IGraph, fileQueries: string[]): Promise<DriveItem[]> {
+  if (!fileQueries || fileQueries.length === 0) {
+    return [];
+  }
+
+  const batch = graph.createBatch();
+  const files = [];
+  const scopes = ['files.read'];
+
+  for (const fileQuery of fileQueries) {
+    if (fileQuery !== '') {
+      batch.get(fileQuery, fileQuery, scopes);
+    }
+  }
+
+  try {
+    const responses = await batch.executeAll();
+
+    for (const fileQuery of fileQueries) {
+      const response = responses.get(fileQuery);
+      if (response && response.content) {
+        files.push(response.content);
+      }
+    }
+
+    return files;
+  } catch (_) {
+    try {
+      return Promise.all(
+        fileQueries
+          .filter(fileQuery => fileQuery && fileQuery !== '')
+          .map(async fileQuery => {
+            const file = await getDriveItemByQuery(graph, fileQuery);
+            if (file) {
+              return file;
+            }
+          })
+      );
+    } catch (_) {
+      return [];
+    }
+  }
+}
+
+async function getDriveItemsByInsights(graph: IGraph, insightResponse, scopes) {
+  if (!insightResponse) {
+    return [];
+  }
+
+  const insightItems = insightResponse.value;
+  const batch = graph.createBatch();
+  const driveItems = [];
+  for (const item of insightItems) {
+    const driveItemId = item.resourceReference.id;
+    if (driveItemId !== '') {
+      batch.get(driveItemId, driveItemId, scopes);
+    }
+  }
+
+  try {
+    const driveItemResponses = await batch.executeAll();
+
+    for (const item of insightItems) {
+      const driveItemResponse = driveItemResponses.get(item.resourceReference.id);
+      if (driveItemResponse && driveItemResponse.content) {
+        driveItems.push(driveItemResponse.content);
+      }
+    }
+    return driveItems;
+  } catch (_) {
+    try {
+      return Promise.all(
+        insightItems
+          .filter(insightItem => insightItem.resourceReference.id && insightItem.resourceReference.id !== '')
+          .map(async insightItem => {
+            const driveItemResponses = await graph
+              .api(insightItem.resourceReference.id)
+              .middlewareOptions(prepScopes(...scopes))
+              .get();
+            if (driveItemResponses && driveItemResponses.length) {
+              return driveItemResponses[0].content;
+            }
+          })
+      );
+    } catch (_) {
+      return [];
+    }
+  }
+}
+
+async function getFilesPageIterator(graph: IGraph, request) {
```

Comment:
This function should return the page iterator instead of fetching all pages. The file list should use the next() method on the page iterator to fetch additional items as needed.